internal/apikey/apikey.go

```diff
@@ -85,19 +88,19 @@ func FromEcdsaPrivateKey(privateKey *ecdsa.PrivateKey) (*ApiKey, error) {
 		return nil, err
 	}
 
-	tkApiKey := ApiKey{
+	return &ApiKey{
 		TkPublicKey:  tkPublicKey,
 		TkPrivateKey: tkPrivateKey,
 		publicKey:    publicKey,
 		privateKey:   privateKey,
-	}
-
-	return &tkApiKey, nil
+	}, nil
 }
 
 // Takes a TK-encoded private key and creates an ECDSA private key
-func FromTkPrivateKey(encodedPrivateKey string) (*ApiKey, error) {
-	bytes, err := hex.DecodeString(encodedPrivateKey)
+func FromTkPrivateKey(encodedPrivateKey []byte) (*ApiKey, error) {
+	bytes := make([]byte, hex.DecodedLen(len(encodedPrivateKey)))
```

Comment:
This is a perfect example here: it's unintuitive for "encodedPrivateKey" to be `[]byte`. Then these bytes are decoded into more bytes.

Reply:
Good point... encoded probably does mean _encoded_. :)